File: examples/go/oracle-price/OraclePriceFeed_test.go
package contract

import (
	"testing"
	runar "github.com/icellan/runar/packages/runar-go"
)

func newOracleFeed() *OraclePriceFeed {
	return &OraclePriceFeed{
		OraclePubKey: runar.RabinPubKey("oracle_rabin_pk"),
		Receiver:     runar.MockPubKey(),
	}
}

func TestOraclePriceFeed_Settle(t *testing.T) {
	newOracleFeed().Settle(60000, runar.RabinSig("sig"), runar.ByteString("pad"), runar.MockSig())
}

func TestOraclePriceFeed_Settle_PriceTooLow_Fails(t *testing.T) {
	defer func() {
		if r := recover(); r == nil {
			t.Fatal("expected assertion failure")
		}
	}()
	newOracleFeed().Settle(50000, runar.RabinSig("sig"), runar.ByteString("pad"), runar.MockSig())
}

func TestOraclePriceFeed_Compile(t *testing.T) {
	if err := runar.CompileCheck("OraclePriceFeed.runar.go"); err != nil {
		t.Fatalf("Rúnar compile check failed: %v", err)
	}
}

File: examples/rust/oracle-price/OraclePriceFeed_test.rs
#[path = "OraclePriceFeed.tsop.rs"]
mod contract;

use contract::*;
use tsop::prelude::*;

fn new_oracle_feed() -> OraclePriceFeed {
    OraclePriceFeed {
        oracle_pub_key: b"oracle_rabin_pk".to_vec(),
        receiver: mock_pub_key(),
    }
}

#[test]
fn test_settle() {
    new_oracle_feed().settle(60000, &b"sig".to_vec(), &b"pad".to_vec(), &mock_sig());
}

#[test]
#[should_panic]
fn test_settle_price_too_low_fails() {
    new_oracle_feed().settle(50000, &b"sig".to_vec(), &b"pad".to_vec(), &mock_sig());
}

#[test]
fn test_settle_high_price() {
    new_oracle_feed().settle(100000, &b"sig".to_vec(), &b"pad".to_vec(), &mock_sig());
}

#[test]
fn test_compile() {
    tsop::compile_check(
        include_str!("OraclePriceFeed.tsop.rs"),
        "OraclePriceFeed.tsop.rs",
    ).unwrap();
}
